pages/first-principles.markdown

```diff
@@ -138,11 +152,15 @@ If instead you model your system as a series of 20,000 LOC modules, with nice AP
 
 On the team organization side, I think it makes a lot of sense to follow an approach like Spotify's squads (see the image [here](https://labs.spotify.com/2014/03/27/spotify-engineering-culture-part-1/), where you setup multiple, sibling teams working on feature teams, and leveraging shared infrastructure that are separate teams and repositories.
 
+:speech_balloon: Hmm, I feel like there is a lot of assumed context here when you talk about teams and repositories. For example, Google [famously has a single repository for all of its code](http://cacm.acm.org/magazines/2016/7/204032-why-google-stores-billions-of-lines-of-code-in-a-single-repository/fulltext), but they are able to run agile teams within that, and I don't think that it's because the overall quality of the codebase is significantly different than at other companies. I think that the most interesting part of this section is your advice to model the system as independent modules; the underlying principle here is that developers need to be empowered to optimize locally. Diffuse ownership and large teams, along with deadlines, is another way that developers end up being disempowered.
```

Comment:
I'll nitpick that Google, yes, has one repository, but they still organize it has a series of flat/sibling projects (basically all of their MPs living in one repo), so the principle of "teams own projects" still basically applies, e.g. yes, I meant "semantic project" and not "literal repository". I'll update accordingly.

I do like the concept and phrasing of local empowerment. I'll work that in somewhere.

This Spotify thing was also tacked on at the end; I'll rework it a bit.